Mon 12658 bbdo3 (#1501)
centreon/centreon-documentation: i18n/fr/docusaurus-plugin-content-docs/version-22.04/developer/developer-broker-mapping.md

      repeated TagInfo tags = 81;
      uint64 icon_id = 82;
    }

### Pb Adaptive host

When BBDO 3 version is used, you can see this event sent when a host has
changes in its configuration.

The [protobuf message](https://developers.google.com/protocol-buffers/docs/proto3)
is the following:

    message AdaptiveHost {
      uint64 host_id = 1;

      optional bool notifications_enabled = 2;
      optional bool active_checks_enabled = 3;
      optional bool should_be_scheduled = 4;
      optional bool passive_checks_enabled = 5;
      optional bool event_handler_enabled = 6;
      optional bool flap_detection_enabled  = 7;
      optional bool obsess_over  = 8;
      optional string event_handler = 9;
      optional string check_command  = 10;
      optional uint32 check_interval  = 11;
      optional uint32 retry_interval  = 12;
      optional uint32 max_check_attempts  = 13;
      optional bool check_freshness = 14;
      optional string check_period  = 15;
      optional string notification_period  = 16;
    }

### Pb Host Status

When BBDO 3 version is used, this type of event is sent instead of
*Host Status*. Its content is almost the same but the old one contains some

> configuration items you don't have here, A *Pb Host Status* is smaller than

Reviewer: ```suggestion
configuration items you don't have here. A *Pb Host Status* is smaller than
```

| host\_id    | unsigned integer | Index ID.   |         |
| service\_id | unsigned integer | Index ID.   |         |

### Pb Rebuild Message

This event comes with BBDO 3. When some graphes have to be rebuild. Messages
handling these rebuilds are of that type. They replace the old bbdo rebuild
message.

There are three states for this message:
* START: here is the first state, this message initializes which metrics have
to be rebuild.
* DATA: once the start STATE has been sent, one or more messages with DATA state
may be sent to the RRD broker.

> * END: When all the rebuild events are sent, this one is sent to close the

Reviewer: ```suggestion
* END: When all the rebuild events have been sent, this one is sent to close the
```

rebuilds. And the RRD broker falls back in a nominal state.

The [protobuf message](https://developers.google.com/protocol-buffers/docs/proto3)
is the following:

    message Point {
      int64 ctime = 1;
      double value = 2;
    }

    message Timeserie {
      repeated Point pts = 1;
      int32 data_source_type = 2;
      uint32 check_interval = 3;
      uint32 rrd_retention = 4;
    }

    message RebuildMessage {
      enum State {
        START = 0;
        DATA = 1;
        END = 2;
      }
      State state = 1;
      /* Only used on DATA state */
      map<uint64, Timeserie> timeserie = 2;

      /* Only used on START/END state */
      repeated uint64 metric_id = 3;
    }

### Pb Remove Graph Message

This event comes with BBDO 3. When we want to remove graph files, we can use

> the centengine gRPC API and this call makes cbd to generate a *Pb Remove Graph

Reviewer: ```suggestion
the centengine gRPC API and this call makes cbd generate a *Pb Remove Graph
```
